### ISSUE TITLE
Error on `Date` type column

### ISSUE THREAD

**Issue by camper42** [housepower/spark-clickhouse-connector#78]
```
CREATE TABLE IF NOT EXISTS default.test_date (`date` Date) Engine Log();
INSERT INTO default.test_date (*) VALUES ('2022-04-12');

# spark sql
spark.sql("select * from `ck-datasupport-1`.dev.test_shard_uid").toPandas()`
```

task failed with error

```
java.lang.ClassCastException: class java.lang.Long cannot be cast to class java.lang.Integer (java.lang.Long and java.lang.Integer are in module java.base of loader 'bootstrap')
	at scala.runtime.BoxesRunTime.unboxToInt(BoxesRunTime.java:103)
	at org.apache.spark.sql.catalyst.expressions.BaseGenericInternalRow.getInt(rows.scala:41)
	at org.apache.spark.sql.catalyst.expressions.BaseGenericInternalRow.getInt$(rows.scala:41)
	at org.apache.spark.sql.catalyst.expressions.GenericInternalRow.getInt(rows.scala:195)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:759)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:349)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:898)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:898)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:131)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:506)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1462)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:509)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

---

the error my colleague encountered was:

```
CREATE TABLE IF NOT EXISTS `default`.`test_uid` (`user_id` INT NOT NULL, `load_date` DATE NOT NULL) ENGINE MergeTree
PARTITION BY load_date
 ORDER BY user_id
SETTINGS index_granularity = '8192'

INSERT INTO `default`.`test_uid` (*) VALUES (1, '2022-04-12');

# spark sql
spark.sql("select * from `ck-datasupport-1`.`default`.`test_uid` ").show()
```

error log:
```
xenon.clickhouse.exception.ClickHouseServerException: [43] DB::Exception: Illegal types of arguments (Date, Int64) of function equals: While processing load_date = ((2022 - 4) - 12)
	at xenon.clickhouse.grpc.GrpcNodeClient.$anonfun$syncStreamQueryAndCheck$3(GrpcNodeClient.scala:193)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:461)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:486)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:492)
	at xenon.clickhouse.format.JSONCompactEachRowWithNamesAndTypesStreamOutput.hasNext(JSONOutputFormat.scala:106)
	at xenon.clickhouse.read.ClickHouseReader.next(ClickHouseReader.scala:71)
	at org.apache.spark.sql.execution.datasources.v2.PartitionIterator.hasNext(DataSourceRDD.scala:93)
	at org.apache.spark.sql.execution.datasources.v2.MetricsIterator.hasNext(DataSourceRDD.scala:130)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:759)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:349)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:898)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:898)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:131)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:506)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1462)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:509)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

**Comment by pan3793** [comment]
Thanks for reporting the issue, will investigate it later

**Comment by pan3793** [comment]
FYI, the fixed version v0.1.1 has been deployed to Maven Central.
```
<dependency>
  <groupId>com.github.housepower</groupId>
  <artifactId>clickhouse-spark-runtime-3.2_2.12</artifactId>
  <version>0.1.0</version>
</dependency>
```

**Comment by camper42** [comment]
still encounters errors on case 2:

`xenon.clickhouse.exception.ClickHouseServerException: [43] DB::Exception: Illegal types of arguments (Date, Int64) of function equals: While processing load_date = ((2022 - 4) - 12)`

**Comment by pan3793** [comment]
The issue should be caused by 
https://github.com/housepower/spark-clickhouse-connector/blob/971d5cf911a80cb41015eef1c25253336573eba2/spark-3.2/clickhouse-spark/src/main/scala/xenon/clickhouse/read/InputPartitions.scala#L43

I'm a little busy now, would you like have a try to fix it?

**Comment by camper42** [comment]
thx for the clue, I'm willing to try (reading code

**Comment by pan3793** [comment]
Fixed by #82
